mysql-test/suite/galera/t/pxc_trx_replay_bf_abort.test

```diff
@@ -0,0 +1,147 @@
+--echo # from Node 2 is allowed to proceed first and it will try to BF-Abort
+--echo # the higher global seqno trx from node1 (certified but not committed yet)
+--echo # Connection node_1
+--connection node_1
+--send COMMIT
+
+--echo # Connection node_1a
+--connection node_1a
+
+--let $galera_sync_point = apply_monitor_slave_enter_sync commit_monitor_master_enter_sync
+--source include/galera_wait_sync_point.inc
+--source include/galera_clear_sync_point.inc
+
+--echo # Let the conflicting UPDATE (from node 2) proceed and wait until it hits abort_trx_end.
+--echo # The victim transaction(node 1) still sits in commit_monitor_master_sync_point.
+
+--let $galera_sync_point = abort_trx_end
+--source include/galera_set_sync_point.inc
+--let $galera_sync_point = apply_monitor_slave_enter_sync
+--source include/galera_signal_sync_point.inc
+--let $galera_sync_point = abort_trx_end commit_monitor_master_enter_sync
+--source include/galera_wait_sync_point.inc
+
+--echo # Let the transactions proceed
+--source include/galera_clear_sync_point.inc
+
+--echo # Node 1 trx will be scheduled for replay and wait just before replaying of trx
+--let $galera_sync_point = start_of_replay_trx
+--source include/galera_set_sync_point.inc
+
+--let $galera_sync_point = abort_trx_end
+--source include/galera_signal_sync_point.inc
+--let $galera_sync_point = commit_monitor_master_enter_sync
+--source include/galera_signal_sync_point.inc
+
+--let $galera_sync_point = start_of_replay_trx
+--source include/galera_wait_sync_point.inc
+
+--echo # Now replaying of trx is paused, at this time, we initiate a trx that acquires
+--echo # conflicting locks required by replay trx
+--echo # Connection node_1b
+--connection node_1b
+UPDATE t1 SET f2 = 'd' WHERE f1 = 1;
+UPDATE t1 SET f2 = 'e' WHERE f1 = 2;
+
+--echo # Let the replay of trx to proceed. This will BF-Abort the new conflicting and
+--echo # not yet committed trx.
+--echo # Connection node_1a
+--connection node_1a
+--let $galera_sync_point = start_of_replay_trx
+--source include/galera_signal_sync_point.inc
+
+--echo # Commit succeeds
+--echo # Connection node_1
+--connection node_1
+--reap
+
+--echo # The new conflicting trx started just before the replay of node1 trx has been
+--echo # BF-Aborted, so we get ER_LOCK_DEADLOCK
+--echo # Connection node_1b
+--connection node_1b
+--error ER_LOCK_DEADLOCK
+COMMIT;
+
+SELECT COUNT(*) = 1 FROM t1 WHERE f2 = 'b';
+SELECT COUNT(*) = 1 FROM t1 WHERE f2 = 'c';
+
+--echo # wsrep_local_replays has increased by 1
+--let $wsrep_local_replays_new = `SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_local_replays'`
+--disable_query_log
+--eval SELECT $wsrep_local_replays_new - $wsrep_local_replays_old = 1 AS wsrep_local_replays;
+--enable_query_log
+
+--echo # Connection node_2
+--connection node_2
+SELECT COUNT(*) = 1 FROM t1 WHERE f2 = 'b';
+SELECT COUNT(*) = 1 FROM t1 WHERE f2 = 'c';
```

Comment:
Can we validate that data is consistent on both servers?

Either this
```
--let $diff_servers = 1 2 
--source include/diff_servers.inc
```

Or this
```
--let $galera_diff_statement = SELECT * FROM t1;
--source include/galera_diff.inc
```